Sources/Basics/Observability.swift

```diff
@@ -567,14 +598,25 @@ extension ObservabilitySystem {
         self.init(DiagnosticsEngineAdapter(diagnosticEngine: diagnosticEngine))
     }
 
-    private struct DiagnosticsEngineAdapter: ObservabilityHandlerProvider, DiagnosticsHandler {
+    private struct DiagnosticsEngineAdapter: ObservabilityHandlerProvider, ScopeDiagnosticsHandler, ScopeOutputHandler, ScopeProgressHandler {
         let diagnosticEngine: DiagnosticsEngine
 
-        var diagnosticsHandler: DiagnosticsHandler { self }
+        var diagnosticsHandler: ScopeDiagnosticsHandler { self }
+        var outputHandler: ScopeOutputHandler { self }
+        var progressHandler: ScopeProgressHandler { self }
 
         func handleDiagnostic(scope: ObservabilityScope, diagnostic: Diagnostic) {
             diagnosticEngine.emit(.init(diagnostic))
         }
+
+        #warning("FIXME: do something here?")
+        func handleOutput(scope: ObservabilityScope, output: String) {
+            // NOOP
```

Comment:
It seems fine to leave this empty for the DiagnosticsEngine compatibility and instead implement output handers in the right places where we want output.